src/compiler/crystal/macros/methods.cr

```diff
@@ -947,6 +953,16 @@ module Crystal
   class NamedTupleLiteral
     def interpret(method : String, args : Array(ASTNode), named_args : Hash(String, ASTNode)?, block : Crystal::Block?, interpreter : Crystal::MacroInterpreter)
       case method
+      when "delete"
+        interpret_one_arg_method(method, args) do |key|
+          unless key.is_a? Crystal::SymbolLiteral | Crystal::StringLiteral
+            key.raise "argument to NamedTupleLiteral#delete must be a string or symbol, not #{key.class_desc}:\n\n#{key}"
+          end
+
+          entries.reject! { |e| e.key == key.value }
```

Comment:
Apparently CI isn't happy about this?  Pretty sure the compiler should be able to see that `key` can only be one of those two types; of which both have a `value` method.

Reply:
Maybe some change in behavior since I started this PR.  Pushing a commit now to fix.